### DISCUSSION TITLE
How set headers cache-control from public files

### DISCUSSION THREAD

**Discussion by huggler** [vercel/next.js#16118] [Help] [answered]
my public directory is

public/assets

and my next.config.js  

"headers": [
    {
      "source": "/assets/*",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "public, max-age=31536000, stale-while-revalidate"
        }
      ]
    }

Error! Header at index 1 has invalid `source` pattern "/assets/*"

**Comment by ijjk** [comment, marked as answer]
Hi, to add the header for all files under the `assets` path you need to use either an un-named catch-all `/assets/(.*)` or named `/assets/:path*`. See the documentation [here](https://nextjs.org/docs/api-reference/next.config.js/headers) for more info

```js
module.exports = {
  headers() {
    return [
      {
        source: "/assets/:path*",
        headers: [
          {
            key: "Cache-Control",
            value: "public, max-age=31536000, stale-while-revalidate",
          },
        ],
      },
    ];
  },
};
```

**Reply by luukdv** [reply]
For future readers: seems like this is not allowed anymore unfortunately (https://nextjs.org/docs/api-reference/next.config.js/headers#cache-control).

**Reply by cbdeveloper** [reply]
@ijjk 

Any workaround to this?

I have some fonts and static svgs on my `public` folder, and I'm getting this cache behavior in production:

![image](https://user-images.githubusercontent.com/43407798/209281532-b07dcb6d-9d80-4dab-9b36-2b683f5975b3.png)
